docs/fundamentals/runtime-libraries/system-reflection-emit-assemblybuilder.md

```diff
@@ -23,17 +26,112 @@ Dynamic assemblies can be created using one of the following access modes:
 
 The access mode must be specified by providing the appropriate <xref:System.Reflection.Emit.AssemblyBuilderAccess> value in the call to the <xref:System.Reflection.Emit.AssemblyBuilder.DefineDynamicAssembly%2A?displayProperty=nameWithType> method when the dynamic assembly is defined and cannot be changed later. The runtime uses the access mode of a dynamic assembly to optimize the assembly's internal representation.
 
-In .NET Framework, you can save dynamic assemblies to files. This feature is not available in .NET Core and .NET 5 and later versions. For an alternative way to generate assembly files, see <xref:System.Reflection.Metadata.Ecma335.MetadataBuilder>.
-
-In .NET Framework, a dynamic assembly can consist of one or more dynamic modules. If a dynamic assembly contains more than one dynamic module, the assembly's manifest file name should match the module's name that is specified as the first argument to the <xref:System.Reflection.Emit.AssemblyBuilder.DefineDynamicModule%2A> method. In .NET Core and .NET 5+, a dynamic assembly can only consist of one dynamic module.
+The following example demonstrates how to create and run assembly:
+
+```cs
+public void CreateAndRunAssembly(string assemblyPath)
+{
+    AssemblyBuilder ab = AssemblyBuilder.DefineDynamicAssembly(new AssemblyName("MyAssembly"), AssemblyBuilderAccess.Run);
+    ModuleBuilder mob = ab.DefineDynamicModule("MyModule");
+    TypeBuilder tb = mob.DefineType("MyType", TypeAttributes.Public | TypeAttributes.Class);
+    MethodBuilder mb = tb.DefineMethod("SumMethod", MethodAttributes.Public | MethodAttributes.Static,
+                                                                   typeof(int), new Type[] {typeof(int), typeof(int)});
+    ILGenerator il = mb.GetILGenerator();
+    il.Emit(OpCodes.Ldarg_0);
+    il.Emit(OpCodes.Ldarg_1);
+    il.Emit(OpCodes.Add);
+    il.Emit(OpCodes.Ret);
+
+    Type type = tb.CreateType();
+    
+    MethodInfo method = type.GetMethod("SumMethod");
+    Console.WriteLine(method.Invoke(null, new object[] { 5, 10 }));
+}
+```
+
+## Persistable dynamic assemblies in .NET Core
+
+The `AssemblyBuilder.Save` API was not ported to .NET core because of technical difficulties like the implementation heavily depend on windows specific native code that also not ported into .NET Core. When .NET Core developers can only run the generated assemblies it was very difficult to debug in-memory assemblies without having an option to save an assembly. Saving the assembly to a file allows generated assemblies to be verified with tools such as ILVerify, or decompiled and manually examined with tools such as ILSpy. Furthermore, the saved assembly can be shared or loaded directly which can be used to decrease application startup time.
```

Comment:
```suggestion
The `AssemblyBuilder.Save` API was not ported to .NET (Core) because the implementation depended heavily on Windows-specific native code that also wasn't ported. Because you could only *run* a generated assembly and not *save* it, it was difficult to debug these in-memory assemblies. Saving the assembly to a file lets you verify the generated assembly with tools such as ILVerify, or decompile and manually examine it with tools such as ILSpy. Furthermore, the saved assembly can be shared or loaded directly, which can decrease application startup time.
```